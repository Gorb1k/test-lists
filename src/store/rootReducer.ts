import {combineReducers} from "@reduxjs/toolkit";
import itemsSlice from "./items/itemsSlice";


export const rootReducer = combineReducers({
    items:itemsSlice
})