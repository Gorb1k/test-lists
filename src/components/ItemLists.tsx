import {FC, useEffect} from 'react';
import MappedList from "./MappedList";
import OriginList from "./OriginList";
import {useDispatch, useSelector} from "react-redux";
import {AppDispatch} from "../store";
import {fetchItems} from "../store/items/items.thunk";
import {selectItems} from "../store/items/items.selector";


const ItemLists: FC = () => {

    const {mappedItems, originItems, isLoading, error} = useSelector(selectItems)
    const dispatch = useDispatch<AppDispatch>()

    useEffect(() => {
        dispatch(fetchItems('http://5af1eee530f9490014ead8c4.mockapi.io/items'))
    },[])

    if (isLoading) return <h1 style={{padding: '2rem'}}>Loading data...</h1>
    if (error) return <h1 style={{padding: '2rem'}}>{error}</h1>
    return (
        <div className={'list'} data-testid={'lists'}>
            <div>
                <h4>Origin list:</h4>
                <OriginList originItems={originItems}/>
            </div>
            <div>
                <h4>Mapped list:</h4>
                <MappedList mappedItems={mappedItems}/>
            </div>
        </div>

    );
};

export default ItemLists;