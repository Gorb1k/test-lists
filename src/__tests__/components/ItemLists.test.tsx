import {render, screen} from "@testing-library/react";

import ItemLists from "../../components/ItemLists";
import {store} from "../../store";
import {Provider} from "react-redux";

describe('Item lists', () => {
    it('should contain 2 headers', async () => {
        render(
            <Provider store={store}>
                <ItemLists/>
            </Provider>
        )
        const origin = await screen.findByText(/Origin list/i)
        const mapped = await screen.findByText(/Mapped list/i)
        expect(origin).toBeInTheDocument()
        expect(mapped).toBeInTheDocument()
    });

    it('ItemLists snapshot', async () => {
        render(
            <Provider store={store}>
                <ItemLists/>
            </Provider>
        )
        const container = await screen.findByTestId('lists')
        expect(container).toMatchSnapshot()
    });
});