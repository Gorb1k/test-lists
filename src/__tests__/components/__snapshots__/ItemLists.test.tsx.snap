// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item lists ItemLists snapshot 1`] = `
<div
  class="list"
  data-testid="lists"
>
  <div>
    <h4>
      Origin list:
    </h4>
    <ul>
      <li>
        <span>
          List item 1
        </span>
      </li>
      <li>
        <span>
          List item 2
        </span>
      </li>
      <li>
        <span>
          List item 3
        </span>
      </li>
      <li>
        <span>
          List item 4
        </span>
      </li>
      <li>
        <span>
          List item 5
        </span>
      </li>
      <li>
        <span>
          List item 6
        </span>
      </li>
      <li>
        <span>
          List item 7
        </span>
      </li>
      <li>
        <span>
          List item 8
        </span>
      </li>
      <li>
        <span>
          List item 9
        </span>
      </li>
    </ul>
  </div>
  <div>
    <h4>
      Mapped list:
    </h4>
    <ul
      data-testid="list"
    >
      <li>
        <span>
          List item 1
        </span>
        <ul
          data-testid="list"
        >
          <li>
            <span>
              List item 5
            </span>
          </li>
          <li>
            <span>
              List item 6
            </span>
          </li>
          <li>
            <span>
              List item 7
            </span>
            <ul
              data-testid="list"
            >
              <li>
                <span>
                  List item 8
                </span>
              </li>
              <li>
                <span>
                  List item 9
                </span>
              </li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <span>
          List item 2
        </span>
      </li>
      <li>
        <span>
          List item 3
        </span>
      </li>
      <li>
        <span>
          List item 4
        </span>
      </li>
    </ul>
  </div>
</div>
`;
